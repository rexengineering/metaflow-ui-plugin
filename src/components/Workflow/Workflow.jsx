import React from "react";
import PropTypes from "prop-types";
import Task from "../Task/Task.container";
import { selectTask } from "../../store/selectors/rexflow";

function Workflow({ workflowID, className, tasks }) {
  const task = selectTask(workflowID, tasks);
  return (
    <section>
      <Task className={className} task={task} />
    </section>
  );
}

Workflow.propTypes = {
  className: PropTypes.string.isRequired,
  workflowID: PropTypes.string.isRequired,
};

export default Workflow;
