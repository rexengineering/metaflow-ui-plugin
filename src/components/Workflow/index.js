export { default } from "./Workflow.container";
