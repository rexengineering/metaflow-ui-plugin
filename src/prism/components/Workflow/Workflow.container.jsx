import { connect } from 'react-redux';
import Workflow from "./Workflow";

const mapStateToProps = (state) => {
    const { rexFlow } = state;
    const { tasks } = rexFlow;
    return {
        tasks
    }
};

export default connect(mapStateToProps)(Workflow);