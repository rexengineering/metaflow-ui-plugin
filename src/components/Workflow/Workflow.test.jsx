import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import Workflow from "./Workflow";
import getStore from "../../store";

describe("<Workflow />", () => {
  const renderWorkflow = (store = getStore({})) => (
    <Provider store={store}>
      <Workflow className="workflow" workflowID="456456" />
    </Provider>
  );

  it("should render", () => {
    const { asFragment } = render(renderWorkflow());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderWorkflow());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });
});
