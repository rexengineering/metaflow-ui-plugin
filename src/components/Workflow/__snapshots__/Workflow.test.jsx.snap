// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workflow /> should match snapshot 1`] = `
<DocumentFragment>
  <section>
    <section
      class="makeStyles-inProgressContainer-9"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 25px; height: 25px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
      <p
        class="MuiTypography-root makeStyles-inProgressMessage-10 MuiTypography-body2"
      >
        Workflow in progress
      </p>
    </section>
  </section>
</DocumentFragment>
`;
